conans/test/integration/package_info_test.py

```diff
@@ -235,3 +237,169 @@ def package_info(self):
         client.run("create conanfile.py dep/1.0@us/ch", assert_error=True)
         self.assertIn("dep/1.0@us/ch package_info(): self.cpp_info.components cannot be used "
                       "with self.cpp_info configs (release/debug/...) at the same time", client.out)
+
+    def package_info_wrong_cpp_info_test(self):
+        conanfile = textwrap.dedent("""
+            import os
+            from conans import ConanFile
+            class Dep(ConanFile):
+                def package_info(self):
+                    self.cpp_info.name = "Boost"
+                    self.cpp_info.components["Accumulators"].includedirs = [os.path.join("boost", 
+                    "accumulators")]
+                    self.cpp_info.libs = ["hello"]
+        """)
+
+        client = TestClient()
+        client.save({"conanfile.py": conanfile})
+        client.run("export . name/1.0@us/ch")  # Does NOT fail on export
+        client.run("create . name/1.0@us/ch", assert_error=True)
+        self.assertIn("name/1.0@us/ch package_info(): self.cpp_info.components cannot be used with "
+                      "self.cpp_info global values at the same time", client.out)
+
+    def package_info_components_complete_test(self):
+        dep = textwrap.dedent("""
+            import os
+            from conans import ConanFile
+            class Dep(ConanFile):
+                exports_sources = "*"
+                def package(self):
+                    self.copy("*")
+                def package_info(self):
+                    self.cpp_info.name = "Galaxy"
+                    self.cpp_info.components["Starlight"].includedirs = [os.path.join("galaxy", "starlight")]
+                    self.cpp_info.components["Starlight"].libs = ["libstarlight"]
+                    self.cpp_info.components["Planet"].includedirs = [os.path.join("galaxy", "planet")]
+                    self.cpp_info.components["Planet"].libs = ["libplanet"]
+                    self.cpp_info.components["Planet"].requires = ["Starlight"]
```

Comment:
We need to think about this syntax as it is not covering a couple of scenarios for the next iteration:
 * `"Starlight"` can refer to another component or to a requirement
 * how can I require a component of a requirement.

Our first thought was something like this, but this is TBD:

![image](https://user-images.githubusercontent.com/1406456/79568131-9d04e100-80b5-11ea-99f9-7ce3a298ef44.png)

here:
 * component `libA` requires package `other` and component `libC` of package `dep`
 * component `libB` requires component `libA` of the same package and component `libC` of package `dep`